swift/Samples/SampleApp/Sources/RootWorkflow.swift

```diff
@@ -0,0 +1,78 @@
+import Workflow
+import WorkflowUI
+import ReactiveSwift
+import Result
+
+
+// MARK: Input and Output
+
+struct RootWorkflow: Workflow {
+    typealias Output = Never
+}
+
+
+// MARK: State and Initialization
+
+extension RootWorkflow {
+
+    enum State {
+        case welcome
+        case demo(name: String)
+    }
+
+    func makeInitialState() -> RootWorkflow.State {
+        return .welcome
+    }
+
+    func workflowDidChange(from previousWorkflow: RootWorkflow, state: inout State) {
+
+    }
+}
+
+
+// MARK: Actions
+
+extension RootWorkflow {
+
+    enum Action: WorkflowAction {
+
+        typealias WorkflowType = RootWorkflow
+
+        case login(name: String)
+
+        func apply(toState state: inout RootWorkflow.State) -> RootWorkflow.Output? {
+
+            switch self {
+            case .login(name: let name):
+                state = .demo(name: name)
+            }
+
+            return nil
+        }
+    }
+}
+
+
+// MARK: Rendering
+
+extension RootWorkflow {
+    typealias Rendering = CrossFadeScreen
+
+    func render(state: RootWorkflow.State, context: RenderContext<RootWorkflow>) -> Rendering {
+        switch state {
+        case .welcome:
+            return CrossFadeScreen(WelcomeWorkflow()
```

Comment:
style nit: As mentioned above I think this’d be better with a named parameter:
```swift
return CrossFadeScreen(
    base: WelcomeWorkflow()
        .mapOutput({ output -> Action in
            switch output {
            case .login(name: let name):
                return .login(name: name)
            }
        })
        .rendered(with: context)))
```

or even if we want to keep it unnamed:
```swift
return CrossFadeScreen(
    WelcomeWorkflow()
        .mapOutput({ output -> Action in
            switch output {
            case .login(name: let name):
                return .login(name: name)
            }
        })
        .rendered(with: context)))
```

Reply:
You are right. I updated to give a name to the first parameter, and it does read better.